src/game/Game.kt

```diff
@@ -46,3 +49,13 @@ private val preventTeleportation: (Move) -> InvalidMoveException? =
         }
 
 
+val preventCrossingBoundaries: (Move) -> InvalidMoveException? =
+        { m: Move ->
+            if (!( Math.abs(m.vector.deltax) >= 0 && Math.abs(m.vector.deltax) >= 0 &&
+                            Math.abs(m.vector.deltax) <= 8 && Math.abs(m.vector.deltay) <= 5))
+                InvalidVectorException() else null
```

Comment:
This assumes a particular worldMap size? Why do we need this at all?

Reply:
right, now I moved the rules to WorldMap and use getNodes to figure out whether the new move is enabling crossing boundaries